tests/properties/test_energies.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+import pytest
+from nomad.datamodel import EntryArchive
+from nomad.units import ureg
+
+from nomad_simulations.schema_packages.physical_property import PhysicalProperty
+from nomad_simulations.schema_packages.properties.energies import (
+    BaseEnergy,
+    KineticEnergy,
+    PotentialEnergy,
+    TotalEnergy,
+)
+
+from . import logger
+
+
+def test_energy_names_and_units_via_values():
+    kin = KineticEnergy(value=1.23 * ureg.joule)
+    pot = PotentialEnergy(value=4.56 * ureg.joule)
+    tot = TotalEnergy(value=5.79 * ureg.joule)
+
+    for sec in (kin, pot, tot):
+        sec.normalize(EntryArchive(), logger)
+        assert sec.name == sec.__class__.__name__
+        assert hasattr(sec.value, 'magnitude')
+        _ = sec.value.to('joule')
+        assert np.isfinite(sec.value.magnitude)
+
+
+def test_total_energy_contributions_basic_flow_and_idempotency():
+    kin = KineticEnergy(value=2.0 * ureg.joule)
+    pot = PotentialEnergy(value=3.0 * ureg.joule)
+
+    tot = TotalEnergy(value=5.0 * ureg.joule)
+    tot.contributions = [kin, pot]
+
+    assert not tot.m_cache.get('_is_normalized', False)
+
+    tot.normalize(EntryArchive(), logger)
+    assert tot.m_cache.get('_is_normalized', False)
+    assert len(tot.contributions) == 2
+
+    assert kin._is_contribution() is True
+    assert pot._is_contribution() is True
+    assert tot._is_contribution() is False
+
+    figures_before = len(tot.figures)
+    tot.normalize(EntryArchive(), logger)
+    assert len(tot.figures) == figures_before
+
+
+@pytest.mark.parametrize(
+    'set_on_main,set_on_child,expect_error',
+    [
+        (False, False, False),
+        (False, True, False),
+        (True, False, True),
+        (True, True, True),
+    ],
+)
+def test_total_energy_contribution_type_validation(
+    caplog, set_on_main, set_on_child, expect_error
+):
+    kin = KineticEnergy(value=1.0 * ureg.joule)
+    pot = PotentialEnergy(value=2.0 * ureg.joule)
+    if set_on_child:
+        kin.contribution_type = 'kinetic'
+
+    tot = TotalEnergy(value=3.0 * ureg.joule)
+    tot.contributions = [kin, pot]
+
+    if set_on_main:
+        tot.contribution_type = 'invalid_on_main'
+
+    with caplog.at_level('ERROR'):
+        tot.normalize(EntryArchive(), logger)
+
+    has_error = any(
+        'contribution_type set but is not a contribution' in rec.message.lower()
+        for rec in caplog.records
+    )
+    assert has_error is expect_error
```

Comment:
This is already tested for physical properties (`test_contribution_type_validation`).